### DISCUSSION TITLE
How can I parse one line type definition ?

### DISCUSSION THREAD

**Discussion by poljsakerik** [dotansimha/graphql-code-generator#8344] [Q&A] [answered]
Hi, I am working on an extension to the typescript-operations plugin. I have what I want but the resulted types are all in one line.
`export type User = {name: string, surname: string, age: number, birthday: string, profilePicture: any}` 
The typescript plugin format its's exported types like:
```
export type User = {
  name: string,
  surname: string,
  age: number,
  birthday: string,
  profilePicture: any
}
```
Is there an already implemented function anywhere that returns the formated string ?

**Comment by charlypoly** [comment, marked as answer]
Hi @poljsakerik,

GraphQL Code Generator does not use any external tool to format the generated TypeScript code.
However, we created a set of utils to make the code generation easier, [like `DeclarationBlock`](https://github.com/dotansimha/graphql-code-generator/blob/d84afec0950b234cd716ea76436e3800a978328c/packages/plugins/other/visitor-plugin-common/src/utils.ts#L110), that you can see used here: https://github.com/dotansimha/graphql-code-generator/blob/aa0b90616f14e16c0708f96dcb580e7297a1bfe9/packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts#L407.

Please note that we handle indenting and line breaks manually, as showcases here: https://github.com/dotansimha/graphql-code-generator/blob/aa0b90616f14e16c0708f96dcb580e7297a1bfe9/packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts#L423
